eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -74,23 +83,52 @@ jobs:
   variables:
     CMOCKA_XML_FILE: "%g-test-results.xml"
     CMOCKA_MESSAGE_OUTPUT: "xml"
+    BuildArgs: ""
 
   steps:
   - checkout: self
     submodules: recursive
 
+  # Install apt dependencies (if appropriate)
+  - bash: sudo apt install -y $(AptDependencies)
+    condition: and(succeededOrFailed(), ne(variables['AptDependencies'], ''))
+    displayName: Install dependencies from apt
+
   - template: /eng/pipelines/templates/steps/vcpkg.yml
     parameters:
       DependenciesVariableName: VcpkgInstall
 
+  - script: |
+      dotnet tool install -g dotnet-reportgenerator-globaltool
+      dotnet tool install dotnet-reportgenerator-globaltool --tool-path tools
+    displayName: Install coverage tools
+    condition: and(succeededOrFailed(), ne(variables['CODE_COVERAGE'], ''))
+
   - template: /eng/pipelines/templates/steps/cmake-build.yml
     parameters:
-      GenerateArgs: $(CmakeArgs)
+      GenerateArgs: "$(CmakeArgs)"
+      BuildArgs: "$(BuildArgs)"
 
   - script: ctest -C Debug -V --tests-regex ${{ parameters.CtestRegex }}
     workingDirectory: build
     displayName: ctest
 
+  # Make coverage targets (specified in coverage_targets.txt) and assemble
+  # coverage report
+  - bash: |
+      make `cat ${{ parameters.ServiceDirectory }}-targets-coverage.txt`
+      ../tools/reportgenerator "-reports:sdk/*/*/*cov_xml.xml" "-targetdir:." "-reporttypes:Cobertura"
+    workingDirectory: build
+    displayName: Generate Code Coverage Data
+    condition: and(succeededOrFailed(), ne(variables['CODE_COVERAGE'], ''))
+  
+  - task: PublishCodeCoverageResults@1
+    inputs:
+      codeCoverageTool: Cobertura
```

Comment:
Add this tool as an internal dependency to cgmanifest

Reply:
we don't use this tool.
We are just telling Azure DevOps that the XML report we are uploading is a `Cobertura` formatted so the Azure DevOps understand it.
We use `gcovr` (already in the cgmanifest) to generate this XML report.